### PR TITLE
deterministic leader selection

### PR DESCRIPTION
**Pull request by alonmuroch**

### CODE REVIEW

File: ibft/leader/round_robin.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package leader
+
+// Round robin leader selection is a fair and sequential leader selection.
+// Each instance/ round change the next leader is selected one-by-one.
```
Comment (vinipx) [inline ibft/leader/round_robin.go:4]: extra space - instance/round change, maybe?

File: ibft/ibft_test.go [hunk 2]
```diff
@@ -32,20 +41,10 @@ func TestIBFT(t *testing.T) {
...
-		//	Consensus: weekday.New(),
-		//	Params:    params,
-		//}))
-		//instances[i].StartEventLoop()
-		//instances[i].StartMessagePipeline()
 	}
 
 	// start repeated timer
-	ticker := time.NewTicker(10 * time.Second)
+	ticker := time.NewTicker(20 * time.Second)
```
Comment (vinipx) [inline ibft/ibft_test.go:47]: minor for learning purposes: why to double the timer?

File: ibft/instance.go [hunk 3]
```diff
@@ -35,6 +38,7 @@ type Instance struct {
 	State            *proto.State
 	network          network.Network
 	Consensus        dataval.Validator
+	LeaderSelection  leader.Selector
```
Comment (vinipx) [inline ibft/instance.go:41]: why not LeaderSelector? is it to differentiate from InstanceOptions?

File: ibft/proto/params.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func (keys PubKeys) Aggregate() bls.PublicKey {
 //DefaultConsensusParams returns the default round change duration time
 func DefaultConsensusParams() *ConsensusParams {
 	return &ConsensusParams{
-		RoundChangeDuration: int64(time.Second * 5),
+		RoundChangeDuration: int64(time.Second * 3),
```
Comment (vinipx) [inline ibft/proto/params.go:25]: not risky? I'm not sure about time limits

File: ibft/ibft_test.go (hunk 2 above)
Comment (alonmuroch) [reply ibft/ibft_test.go]: to see failures of consensus

File: ibft/proto/params.go (hunk 4 above)
Comment (alonmuroch) [reply ibft/proto/params.go:25]: let's leave it for now as is

**Pull request merged**